backend/src/agents/chart_generator_agent.py

```diff
@@ -0,0 +1,99 @@
+import logging
+from src.prompts import PromptEngine
+from .agent import Agent, agent
+from .tool import tool
+from .agent_types import Parameter
+from src.utils.log_publisher import LogPrefix, publish_log_info
+import os
+import base64
+from src.websockets.connection_manager import connection_manager
+
+logger = logging.getLogger(__name__)
+
+engine = PromptEngine()
+
+async def generate_chart(question_intent, categorical_values, question_params, timeframe, llm, model) -> str:
+    details_to_generate_chart_code = engine.load_prompt(
+        "details-to-generate-chart-code",
+        question_intent=question_intent,
+        categorical_values=categorical_values,
+        question_params=question_params,
+        timeframe=timeframe
+    )
+
+    generate_chart_code_prompt = engine.load_prompt("generate-chart-code")
+    generated_code = await llm.chat(model, generate_chart_code_prompt, details_to_generate_chart_code)
+    sanitised_script = sanitise_script(generated_code)
+    logger.info(f"Sanitised script: {sanitised_script}")
+    await publish_log_info(LogPrefix.USER, f"Code generated by the LLM and sanitised: {sanitised_script}", __name__)
+
+    try:
+        local_vars = {}
+        exec(sanitised_script, {}, local_vars)
+        fig = local_vars.get('fig')
+        if fig is None:
+            raise ValueError("The generated code did not produce a figure named 'fig'.")
+        output_dir = '/app/output'
+        if not os.path.exists(output_dir):
+            os.makedirs(output_dir)
+        logger.info("Saving the figure as 'output.png'")
+        output_path = os.path.join(output_dir, "output.png")
+        fig.savefig(output_path)
+        logger.info(f"Figure saved successfully as {output_path}")
+
+        with open(output_path, "rb") as image_file:
+            image_data = base64.b64encode(image_file.read()).decode("utf-8")
+
+        await connection_manager.send_chart({"type": "image", "data": image_data})
```

Comment:
Should the agent be sending the chart to the frontend, or, should that be the job of the answer agent, meaning we should probably put the chart on the scratchpad.